src/menu/onboarding.c

```diff
@@ -0,0 +1,147 @@
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "onboarding.h"
+
+#include "constant_texts.h"
+#include "host_interface.h"
+#include "menu_priv.h"
+#include "status_api.h"
+#include "ui_delay.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+typedef struct {
+  bool static_screen;
+  bool update_required;
+} onboarding_ctx_t;
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+static onboarding_ctx_t onboarding_ctx = {.static_screen = false,
+                                          .update_required = true};
+
+static const flow_step_t onboarding_flow = {
+    .step_init_cb = onboarding_initialize,
+    .p0_cb = NULL,
+    .ui_cb = NULL,
+    .usb_cb = host_interface,
+    .nfc_cb = NULL,
+    .evt_cfg_ptr = &main_menu_evt_config,
+    .flow_data_ptr = NULL};
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+void onboarding_initialize(engine_ctx_t *ctx, const void *data_ptr) {
```

Comment:
add this function to onboarding_priv.h for consistency

Reply:
Updated